tests/CommandLine.Tests/Unit/Issue482Tests.cs

```diff
@@ -0,0 +1,197 @@
+                "-f, --foxtrot",
+                "--help           Display this help screen.",
+                "--version        Display version information.",
+                "value pos. 0"
+            };
+            Assert.Equal(expected.Count, helps.Count);
+            int i = 0;
+            foreach (var expect in expected)
+            {
+                Assert.Equal(expect.Trim(), helps[i].Trim());
+                i++;
+            }
+
+            ;
+        }
+
+        [Fact]
+        public void AutoBuild_with_ordering_on_shortName()
+        {
+            string expectedCompany = "Company";
+
+
+            var parser = Parser.Default;
+            var parseResult = parser.ParseArguments<Options_HelpText_Ordering_Verb1, Options_HelpText_Ordering_Verb2>(
+                    new[] { "verb1", "--alpha", "alpaga", "--alpha2", "alala", "--charlie", "charlot" })
+                .WithNotParsed(errors => { throw new InvalidOperationException("Must be parsed."); })
+                .WithParsed(args => {; });
+
+            Comparison<ComparableOption> orderOnShortName = (ComparableOption attr1, ComparableOption attr2) =>
+                   {
+                       if (attr1.IsOption && attr2.IsOption)
+                       {
+                           if (attr1.Required && !attr2.Required)
+                           {
+                               return -1;
+                           }
+                           else if (!attr1.Required && attr2.Required)
+                           {
+                               return 1;
+                           }
+                           else
+                           {
+                               if (string.IsNullOrEmpty(attr1.ShortName) && !string.IsNullOrEmpty(attr2.ShortName))
+                               {
+                                   return 1;
+                               }
+                               else if (!string.IsNullOrEmpty(attr1.ShortName) && string.IsNullOrEmpty(attr2.ShortName))
+                               {
+                                   return -1;
+                               }
+                               return String.Compare(attr1.ShortName, attr2.ShortName, StringComparison.Ordinal);
+                           }
+                       }
+                       else if (attr1.IsOption && attr2.IsValue)
+                       {
+                           return -1;
+                       }
+                       else
+                       {
+                           return 1;
+                       }
+                   };
+
+            string message = HelpText.AutoBuild(parseResult,
+                    error =>
+                    {
+                        error.OptionComparison = orderOnShortName;
+                        return error;
+                    },
+                    ex => ex,
+                        false,
+                        80,
+                        orderOnShortName);
```

Comment:
//remove that line
remove the line orderOnShortName
Enough to call without extra parameters:

            string message = HelpText.AutoBuild(parseResult,
                    error =>
                    {
                        error.OptionComparison = orderOnShortName;
                        return error;
                    },
                    ex => ex);